src/seedu/addressbook/data/person/Address.java

```diff
@@ -3,54 +3,70 @@
 import seedu.addressbook.data.exception.IllegalValueException;
 
 /**
- * Represents a Person's address in the address book.
- * Guarantees: immutable; is valid as declared in {@link #isValidAddress(String)}
+ * Represents a Person's address in the address book. Guarantees: immutable; is
+ * valid as declared in {@link #isValidAddress(String)}
  */
 public class Address {
 
-    public static final String EXAMPLE = "123, some street";
-    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
+    public static final String EXAMPLE = "123, Clementi Ave 3, #12-34, 231534";
+    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses must be in form of <BLOCK, STREET, UNIT, POSTAL_CODE>";
     public static final String ADDRESS_VALIDATION_REGEX = ".+";
 
-    public final String value;
+    public static final int ADDRESS_DATA_INDEX_BLOCK = 0;
+    public static final int ADDRESS_DATA_INDEX_STREET = 1;
+    public static final int ADDRESS_DATA_INDEX_UNIT = 2;
+    public static final int ADDRESS_DATA_INDEX_POSTALCODE = 3;
+    public static final String ADDRESS_DATA_SPLITTER = ", ";
+
+    private final Block block;
+    private final Street street;
+    private final Unit unit;
+    private final PostalCode postalCode;
     private boolean isPrivate;
 
     /**
      * Validates given address.
      *
-     * @throws IllegalValueException if given address string is invalid.
+     * @throws IllegalValueException
+     *             if given address string is invalid.
      */
     public Address(String address, boolean isPrivate) throws IllegalValueException {
         String trimmedAddress = address.trim();
         this.isPrivate = isPrivate;
         if (!isValidAddress(trimmedAddress)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = trimmedAddress;
+        String[] splittedAddress = address.split(ADDRESS_DATA_SPLITTER);
+        this.block = new Block(splittedAddress[ADDRESS_DATA_INDEX_BLOCK],isPrivate);
```

Comment:
Think why it makes little sense to have individual components marked as private or not (at least in the context of this application)?